src/Features/CSharp/Portable/Completion/CompletionProviders/CompletionUtilities.cs

```diff
@@ -68,6 +69,8 @@ internal static bool IsTriggerCharacter(SourceText text, int characterPosition,
             return false;
         }
 
+        internal static ImmutableHashSet<char> PossibleTriggerCharacters => ImmutableHashSet.Create('.', '#', '>', ':');
```

Comment:
1. consider naming this CommonTriggercharacters.  
2. Seems odd to  have `#`, and `:` in here.  When do those trigger?  can those be in their own specific providers?

Reply:
1.  Sure, that sounds better.
2.  I used the characters from the IsTriggerCharacter above.  So every provider that uses IsTriggerCharacter (most of them) has this set of possible completion characters.  This set is supposed to match IsTriggerCharacer so I can just drop this in wherever a provider uses IsTriggerCharacter.